lib/ansible/plugins/callback/logdna.py

```diff
@@ -0,0 +1,214 @@
+            key: plugin_ignore_errors
+        default: False
+      conf_hostname:
+        required: False
+        description: Alternative Host Name; the current host name by default
+        type: string
+        env:
+          - name: LOGDNA_HOSTNAME
+        ini:
+          - section: callback_logdna
+            key: conf_hostname
+      conf_tags:
+        required: False
+        description: Tags
+        type: string
+        env:
+          - name: LOGDNA_TAGS
+        ini:
+          - section: callback_logdna
+            key: conf_tags
+'''
+
+import logging
+import json
+import socket
+from uuid import getnode
+from ansible.plugins.callback import CallbackBase
+
+try:
+    from logdna import LogDNAHandler
+    HAS_LOGDNA = True
+except ImportError:
+    HAS_LOGDNA = False
+
+
+# Getting MAC Address of system:
+def get_mac():
+    mac = "%012x" % getnode()
+    return ":".join(map(lambda index: mac[index:index + 2], range(int(len(mac) / 2))))
+
+
+# Getting hostname of system:
+def get_hostname():
+    return str(socket.gethostname()).split('.local')[0]
+
+
+# Getting IP of system:
+def get_ip():
+    try:
+        return socket.gethostbyname(get_hostname())
+    except BaseException:
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        try:
+            s.connect(('10.255.255.255', 1))
+            IP = s.getsockname()[0]
+        except BaseException:
+            IP = '127.0.0.1'
+        finally:
+            s.close()
+        return IP
+
+
+# Is it JSON?
+def isJSONable(obj):
+    try:
+        json.dumps(obj)
+        return True
+    except BaseException:
+        return False
+
+
+# LogDNA Callback Module:
+class CallbackModule(CallbackBase):
+
+    CALLBACK_VERSION = 0.1
+    CALLBACK_TYPE = 'aggregate'
+    CALLBACK_NAME = 'logdna'
+    CALLBACK_NEEDS_WHITELIST = True
+
+    def __init__(self, display=None):
+        super(CallbackModule, self).__init__(display=display)
+
+        self.disabled = True
+        self.playbook_name = None
+        self.playbook = None
+        self.conf_key = None
+        self.plugin_ignore_errors = None
+        self.conf_hostname = None
+        self.conf_tags = None
+
+    def set_options(self, task_keys=None, var_options=None, direct=None):
+        super(CallbackModule, self).set_options(task_keys=task_keys, var_options=var_options, direct=direct)
+
+        self.conf_key = self.get_option('conf_key')
+        self.plugin_ignore_errors = self.get_option('plugin_ignore_errors')
+        self.conf_hostname = self.get_option('conf_hostname')
+        self.conf_tags = self.get_option('conf_tags')
+        self.mac = get_mac()
+        self.ip = get_ip()
+
+        if self.conf_hostname is None:
+            self.conf_hostname = get_hostname()
+
+        if self.conf_tags is None:
+            self.conf_tags = ['ansible']
+        else:
+            self.conf_tags = self.conf_tags.split(',')
+
+        if self.conf_key is None:
+            self.disabled = True
```

Comment:
again, this is not required, since the key IS REQUIRED you never get to this point